scipy/sparse/_base.py

```diff
@@ -656,26 +716,36 @@ def __pow__(self, other):
 
             if other == 0:
                 from ._construct import eye
-                return eye(self.shape[0], dtype=self.dtype)
+                return eye(
+                    self.shape[0], dtype=self.dtype, _array=self._is_array
+                )
             elif other == 1:
                 return self.copy()
             else:
                 tmp = self.__pow__(other//2)
                 if (other % 2):
-                    return self * tmp * tmp
+                    return self @ tmp @ tmp
                 else:
-                    return tmp * tmp
+                    return tmp @ tmp
         elif isscalarlike(other):
             raise ValueError('exponent must be an integer')
         else:
             return NotImplemented
 
     def __getattr__(self, attr):
         if attr == 'A':
+            if self._is_array:
+                warn(np.VisibleDeprecationWarning(
+                    "Please use `.todense()` instead"
```

Comment:
I think the plan is to recommend `.toarray()` overall.

Reply:
`.todense` on a sparse array currently produces a dense array. This is consistent with `.todense` on a sparse matrix producing a dense matrix. Because the name is easier to understand than `.toarray` (huh, don't we have an array already?), I think it makes sense to advertise `.todense` instead of `.toarray`.